pkg/server/api_v2_grants.go

```diff
@@ -0,0 +1,365 @@
+		sessiondata.NodeUserSessionDataOverride, `
+SELECT name FROM system.namespace
+WHERE "parentID" = 0 AND "parentSchemaID" = 0 AND id = $1`, dbId)
+	if err != nil {
+		return resp, err
+	}
+
+	if row == nil || row[0] == tree.DNull {
+		// Database id does not exist.
+		return resp, nil
+	}
+
+	dbName := string(*row[0].(*tree.DString))
+	resp.Name = dbName
+
+	escDbName := tree.NameStringP(&dbName)
+
+	query := safesql.NewQuery()
+	query.Append(fmt.Sprintf(`
+SELECT grantee, privilege_type, count(*) OVER() as total_row_count
+FROM [SHOW GRANTS ON DATABASE %s]`, escDbName))
+
+	// Pagination arguments.
+	if limit > 0 {
+		query.Append(" LIMIT $", limit)
+	}
+	if offset > 0 {
+		query.Append(" OFFSET $", offset)
+	}
+
+	resp.Results = make([]grantRecord, 0)
+	it, err := ie.QueryIteratorEx(ctx, "get-database-grants", nil, /* txn */
+		sessiondata.InternalExecutorOverride{User: sqlUser},
+		query.String(), query.QueryArguments()...,
+	)
+	if err != nil {
+		return resp, err
+	}
+
+	defer func(it isql.Rows) {
+		retErr = errors.CombineErrors(retErr, it.Close())
+	}(it)
+
+	ok, err := it.Next(ctx)
+
+	if err != nil || !ok {
+		// If ok is false, the query returned 0 rows.
+		return resp, nil
+	}
+
+	scanner := makeResultScanner(it.Types())
+	for ; ok; ok, err = it.Next(ctx) {
+		if err != nil {
+			return resp, err
+		}
+		grant := grantRecord{}
+		row := it.Cur()
+		if resp.PaginationInfo.TotalResults == 0 {
+			resp.PaginationInfo.TotalResults = int64(tree.MustBeDInt(row[2]))
+		}
+		if err := scanner.Scan(row, "grantee", &grant.Grantee); err != nil {
+			return resp, err
+		}
+		if err := scanner.Scan(row, "privilege_type", &grant.Privilege); err != nil {
+			return resp, err
+		}
+		resp.Results = append(resp.Results, grant)
+	}
+	return resp, nil
+}
+
+// getTableGrants returns a paginated response of table grants metadata.
+// If the user does not have the required permissions to access the table, we will return a 403 Forbidden error.
+//
+// ---
+// parameters:
+//   - name: tableId
+//     type: integer
+//     description: The ID of the table to get grants for.
+//     in: path
+//
+// produces:
+// - application/json: tableGrantsResponse
+//
+// responses:
+//
+//	"200":
+//		description: A paginated response of grants on the provided table.
+//	"404":
+//		description: The table does not exist.
+//	"400":
+//		description: The request is malformed.
+//	"500":
+//		description: An internal server error occurred.
+func (a *apiV2Server) getTableGrants(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	ctx = a.sqlServer.AnnotateCtx(ctx)
+	sqlUser := authserver.UserFromHTTPAuthInfoContext(ctx)
+	if r.Method != http.MethodGet {
+		http.Error(w, "Method Not Allowed", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Validate request parameters.
+	pathVars := mux.Vars(r)
+	tableId, err := strconv.Atoi(pathVars[tableIdPathVar])
+	if err != nil || tableId < 0 {
+		http.Error(w, "invalid database id", http.StatusBadRequest)
+		return
+	}
```

Comment:
Same question here as in database grants

Reply:
Done.